File: Day300/maximumElementAfterDecreasingAndRearranging.java
class Solution {
    public int maximumElementAfterDecrementingAndRearranging(int[] arr) {
        int l = arr.length;
        int temp[] = new int[l];
        for (int i=0; i<l; i++) {
            temp[Math.min(arr[i]-1, l-1)]++;
        }   
        int answer = 1;
        for (int i=1; i<l; i++) {
            answer = Math.min(i+1, answer+temp[i]);
        }
        return answer;
    }
}

File: Day300/maximumElementAfterDecreasingAndRearranging.go
func maximumElementAfterDecrementingAndRearranging(arr []int) int {
    sort.Ints(arr)
    arr[0] = 1
    for i:=1; i<len(arr); i++ {
        arr[i] = int(math.Min(float64(arr[i-1]+1), float64(arr[i])))
    }
    return arr[len(arr)-1]
}